scripts/linters/other_files_linter.py

```diff
@@ -193,12 +194,10 @@ def check_third_party_libs_type_defs(
                 # In cases where the version is in the form of git commit hashes
                 # such as 'git+https://github.com/username/repo#commit-hash',
                 # we extract the commit hash and use it as the version.
-                elif lib_version[:3] == 'git':
-                    cntr = 0
-                    while cntr < len(lib_version) and lib_version[cntr] != '#':
-                        cntr += 1
-
-                    lib_version = lib_version[(cntr + 1):]
+                elif lib_version[:23] == 'git+https://github.com/':
```

Comment:
The "23" here is a "magic number" and folks won't know where it comes from. I suggest you define a constant for the 'git+...' string and use len() here when referencing it.